pkg/client/get_options.go

```diff
@@ -20,29 +20,60 @@ import (
 	"github.com/codenotary/immudb/pkg/api/schema"
 )
 
+// GetOption is used to set additional options when reading a value with a Get call
 type GetOption func(req *schema.KeyRequest) error
 
+// NoWait option set to true means that the server should not wait for the indexer
+// to be up-to-date with the most recent transaction.
+//
+// In practice this means that the user will get the result instantly without the
+// risk of the server wait for the indexer that may happen in case of a large spike
+// of new transactions.
+//
+// The disadvantage of using this option is that the server can reply with an older
+// value for the same key or with a key not found result even though already committed
+// transactions contain newer updates to that key.
 func NoWait(nowait bool) GetOption {
 	return func(req *schema.KeyRequest) error {
 		req.NoWait = nowait
 		return nil
 	}
 }
 
+// SinceTx option can be used to avoid waiting for the indexer to be up-to-date
+// with the most recent transaction. The client requires though that at least
+// the transaction with id given in the tx parameter must be indexed.
+//
+// This option can be used to avoid additional latency while the server waits
+// for the indexer to finish indexing but still guarantees that specific portion
+// of the database history has already been indexed.
 func SinceTx(tx uint64) GetOption {
 	return func(req *schema.KeyRequest) error {
 		req.SinceTx = tx
 		return nil
 	}
 }
 
+// AtTx option is used to specify that the value read must be the one set
+// at transaction with id given in the tx parameter.
+//
+// If the key was not modified at given transaction, the request will
+// return key not found result even if the key was changed before that transaction.
+//
+// Using AtTx also allows reading entries set with disabled indexing.
 func AtTx(tx uint64) GetOption {
 	return func(req *schema.KeyRequest) error {
 		req.AtTx = tx
 		return nil
 	}
 }
 
+// AtRevision request specific revision for given key.
```

Comment:
`AtRevision requests a specific revision for a given key.`

wdyt on adding the below to the comment? 
```
// If Revision = 0, returns current value
// If Revision > 0, returns nth revision value
// If Revision < 0, returns nth revision value from the end
```

Reply:
Changed in 3a4f6784

Reply:
awesome